### DISCUSSION TITLE
Framework: Multiple categories created with the same Name & Code

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Knowlg/Community#102] [Bugs]
We are creating new framework for Course workflow. While creating the framework with categories, we executed creation of framework category multiple times(while verifying the postman tests). It's created multiple categories with the same name & code.

**API:** {{host}}/api/framework/v1/read/{{course_framework}}

**Response:**
`{
    "id": "api.framework.read",
    "ver": "1.0",
    "ts": "2023-03-08T05:46:52.026Z",
    "params": {
        "resmsgid": "a03865a0-bd74-11ed-97e3-51934d7d7c4b",
        "msgid": "a035a680-bd74-11ed-bbbf-b9b0a7931a82",
        "status": "successful",
        "err": null,
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "framework": {
            "identifier": "sunbird_k-12",
            "code": "sunbird_k-12",
            "name": "sunbird_k-12",
            "description": "Sunbird k-12 framework",
            "categories": [
                {
                    "identifier": "sunbird_k-12_gradelevel",
                    "code": "gradeLevel",
                    "translations": null,
                    "name": "GradeLevel",
                    "description": null,
                    "index": 1,
                    "status": "Live"
                },
                {
                    "identifier": "sunbird_k-12_medium",
                    "code": "medium",
                    "translations": null,
                    "name": "Medium",
                    "description": null,
                    "index": 2,
                    "status": "Live"
                },
                {
                    "identifier": "sunbird_k-12_subject",
                    "code": "subject",
                    "translations": null,
                    "name": "Subject",
                    "description": null,
                    "index": 3,
                    "status": "Live"
                },
                {
                    "identifier": "sunbird_k-12_subject",
                    "code": "subject",
                    "translations": null,
                    "name": "Subject",
                    "description": null,
                    "index": 4,
                    "status": "Live"
                },
                {
                    "identifier": "sunbird_k-12_subject",
                    "code": "subject",
                    "translations": null,
                    "name": "Subject",
                    "description": null,
                    "index": 5,
                    "status": "Live"
                },
                {
                    "identifier": "sunbird_k-12_subject",
                    "code": "subject",
                    "translations": null,
                    "name": "Subject",
                    "description": null,
                    "index": 6,
                    "status": "Live"
                }
            ],
            "type": "TPD",
            "objectType": "Framework"
        }
    }
}`

**Snapshot for reference:**

<img width="1315" alt="Screenshot 2023-03-08 at 11 17 41 AM" src="https://user-images.githubusercontent.com/4496096/223630281-190abbce-97af-4660-86a8-73efecfaea20.png">

@pallakartheekreddy @Krishnaj20  @gandham-santhosh  @vrayulu

**Comment by pallakartheekreddy** [comment]
@vinukumar-vs We need to enable neo4j unique key constraint for neo4j post neo4j deployment. Please refer below automation script
https://github.com/project-sunbird/sunbird-learning-platform/blob/832996fc03f6ca64f6fb5e3f1e4e8aa54fda31f8/ansible/roles/neo4j-deploy/tasks/main.yml#L43

**Reply by vinukumar-vs** [reply]
While setting up this, we have to remove all the existing duplicate nodes in the Neo4J first & then setup the Unique key constraint key as you mentioned.

@gandham-santhosh @Krishnaj20

**Reply by pallakartheekreddy** [reply]
@vinukumar-vs As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
